### PR TITLE
Add feature: Click by string resource

### PR DESCRIPTION
**Pull request by alorma**
As issue:  [#71](https://github.com/SchibstedSpain/Barista/issues/71) points, we cannot click on a view wusing `R.sting.xxx` resources.

This PR fixes it!

### CODE REVIEW

File: library/src/main/java/com/schibsted/spain/barista/BaristaClickActions.java [hunk 1]
```diff
@@ -15,17 +18,43 @@
 
 public class BaristaClickActions {
 
-  public static void click(@IdRes int id) {
+  private static final ResourceTypeChecker RESOURCE_TYPE_CHECKER = new ResourceTypeChecker();
```
Comment (Sloy) [inline library/src/main/java/com/schibsted/spain/barista/BaristaClickActions.java]: It's probably a good idea to just make the methods static. Having an instance doesn't add any value, and we're doing this ugly thing in two places already.
Not sure if that belongs to this PR though.

File: library/src/main/java/com/schibsted/spain/barista/BaristaClickActions.java [hunk 2]
```diff
@@ -15,17 +18,43 @@
...
       try {
-        scrollAndClickDisplayedView(id);
+        scrollAndClickDisplayedViewByString(id);
       } catch (PerformException parentIsNotAnScrollView) {
-        clickDisplayedView(id);
+        clickDisplayedViewByString(id);
       }
     } catch (PerformException parentIsNotAnScrollView) {
-      clickDisplayedView(id);
+      clickDisplayedViewByString(id);
```
Comment (Sloy) [inline library/src/main/java/com/schibsted/spain/barista/BaristaClickActions.java]: Hmm with this change each click method is not triplicated.
I think it's better to extract a method that receives a matcher, and pass the matcher from the generic method (`withText("")`, `withText(res)`, `withId(id)`).

Did I explained myself good enough? What do you think?

Comment (alorma) [inline library/src/main/java/com/schibsted/spain/barista/BaristaClickActions.java]: Uhmm, not understood you :'

**Comment by alorma** [comment]
@rocboronat i will add on README.md as soon as i get it working in all situations, and fixed @Sloy comments

**Comment by rocboronat** [comment]
Ok @alorma :·

**Comment by alorma** [comment]
HI @Sloy i tried to implement your point, but then 2 tests fail (related with viewPager), could you hel with it?

**Comment by Sloy** [comment]
@alorma yes, I think I know the issue. Let me try it and push some changes if it works.

**Comment by alorma** [comment]
Zanku! 😘

**Comment by Sloy** [comment]
There you go @alorma, with that small change tests are passing again.

I think the code can be simplified even more, but that's not related with the changes on this PR.

**Comment by alorma** [comment]
Readme.md updated @Sloy 

Waiting for @rocboronat approve

File: README.md [hunk 3]
```diff
@@ -16,6 +16,7 @@ Barista introduces a discoverable API for the Espresso features. So, you and all
 ```java
 // Click widgets
 click(R.id.button);
+click(R.stiring.button_text); // This will click a view that uses `<string name="button_text">Button text</string>`
```
Comment (rocboronat) [inline README.md]: There's a typo: `stiring`. In additon, we're introducing Barista to developers. Let's avoid overexplaining things like this... I mean, `click(R.string.something)` is going to click the view with the text `R.string.something` for sure.

**Comment by alorma** [comment]
Fixed with amend commit! Thanks for typo!

**Pull request merged**